version-control/set-up/tutorial.md

```diff
@@ -0,0 +1,165 @@
+
+![RSA Keys Generated](images/setup-rsa-key-generated.png)
+
+**# Add SSH Keys to the ssh-agent**
+
+To avoid having to provide a password every time we connect to a service we can use the ssh-agent on our system. This agent handles passwords for ssh private keys and connections to the remote service we're logging into. This avoids having to send passwords over the network to the service.
+
+Check if the ssh-agent running by typing the following command
+
+```bash
+eval "$(ssh-agent -s)"
+```
+This should return the PID for the ssh-agent which means it has been started in the background.
+
+![PID Number](images/setup-pid-number.png)
+
+Now add the ssh keys to the ssh-agent by typing one of the following depending on your system.
+```bash
+Windows:  ssh-add ~/.ssh/id_rsa 
+    Mac:  ssh-add -K ~/.ssh/id_rsa
+```
+> The -K option is Apple's version of `ssh-add` which adds the passphrase to your keychain for when you add the key to the ssh-agent
+
+You can check what identities have been added to the ssh-agent by typing:
+```bash
+ssh-add -l
+```
+
+## Set-up and Connect to a Github Account
+Now that we have Git working on our local system and the SSH keys we'll need in order to connect to a remote service, let's set-up the Github account so we can then finish connecting the two.
+
+### Sign-up for a Github Account
+Navigate to [https://github.com/](https://github.com/) and from the home page sign-up for an account.
+
+![Sign-up for a Github account](images/setup-get-a-github-account.png)
+
+Be sure to choose the free pricing plan.
+
+![Github Free Pricing Plan](images/setup-choose-pricing-plan.png)
+
+Once you've verified your email address the account will be active and you can go back to edit your profile later on.
+
+### Add the SSH Key to Github
+
+Now we have both Git and the SSH keys set-up on our local system and a Github account created so we can have access to hosted repositories, we need to link them together using the SSH keys.
+
+Navigate back to github.com and ensure you're signed-in to your account. Then in the top right corner click on your profile photo and select `settings`.
+
+![Github Settings](images/setup-github-settings.png)
+
+In the Settings sidebar, select the `SSH and GPG Keys` option.
+
+![SSH option is Settings](images/setup-ssh-key-option.png)
+
+On the right hand side, select `New SSH Key` which will present you with an area to paste your public SSH key text into.
+
+![Add new SSH Key to Github](images/setup-add-new-ssh-key.png)
+
+We need to copy to the clipboard the contents of the `id_rsa_pub` file we created earlier. In a console window type the following command.
+```bash
+pbcopy < ~/.ssh/id_rsa.pub
+```
+Now return back to the broswer where your Github account is open. Give the SSH key a `Title` and paste the file contents into the `Key` field. When done, hit the `Add SSH Key` button. You may be prompted for your password, if so enter it and hit `Confirm Password` to complete this step.
```

Comment:
Little typo here `broswer` should be `browser`

Reply:
doh...fixed.